contracts/Gate.sol

```diff
@@ -217,4 +233,43 @@ contract Gate is IGate, Ownable, Pausable {
         _unpause();
     }
 
+    /**
+     * @notice Checks if user possesses the required balance of the conditional token for given voucher set
+     * @param _user user address
+     * @param _tokenIdSupply an ID of a supply token (ERC-1155) [voucherSetID]
+     * @return true if user possesses conditional token, and the token is not deactivated
+     */
+    function checkBalance(address _user, uint256 _tokenIdSupply)
+        internal
+        view
+        returns (bool)
+    {
+        ConditionalCommitInfo memory conditionalCommitInfo = voucherSetToConditionalCommit[_tokenIdSupply];
+        return
+            !isDeactivated[_user][_tokenIdSupply] &&
+            ((conditionalTokenType == TokenType.NONFUNGIBLE_TOKEN || conditionalTokenType == TokenType.FUNGIBLE_TOKEN)
```

Comment:
Instead of checking id it's `NONFUNGIBLE_TOKEN` or `FUNGIBLE_TOKEN` you could just check if it's `MULTI_TOKEN` and switched what should be executed.

Reply:
Changed. Will be included in next commit.